lldb/source/Plugins/ObjectFile/XCOFF/ObjectFileXCOFF.cpp

```diff
@@ -81,9 +79,44 @@ ObjectFile *ObjectFileXCOFF::CreateInstance(const lldb::ModuleSP &module_sp,
   if (!objfile_up)
     return nullptr;
 
+  // Cache xcoff binary.
+  if (!objfile_up->CreateBinary())
+    return nullptr;
+
+  if (!objfile_up->ParseHeader())
+    return nullptr;
+
   return objfile_up.release();
 }
 
+bool ObjectFileXCOFF::CreateBinary() {
+  if (m_binary)
+    return true;
+
+  Log *log = GetLog(LLDBLog::Object);
+
+  auto binary = llvm::object::XCOFFObjectFile::createObjectFile(
```

Comment:
```suggestion
  auto binary = llvm::object::XCOFFObjectFile::create(
```

And then you don't need the dyn_cast below. (You do need to make the create function public, but I don't think that should be controversial given that it's already public for all other object file types)

Reply:
One concern here is, there are two instances of this: `create` and `createXCOFFObjectFile` both of which are private. 
Also they expect a `TYPE_ID` as a paramater, which is again a protected member of the `Binary` class. 
`createObjectFile` is the only accessible function that accepts a public `file_magic::xcoff_object_64` type of parameter and internally calls the appropriate `create(XCOFF|ELF|others)ObjectFile` APIs and passes thec correct TYPE_ID too.
How shall we deal with it in this scenario?

Reply:
Addressed all the other comments. Please do let me know which one to proceed with, as per the above comment.
Thanks!

Reply:
Yes, that will be good. In that case, I can open a PR under llvm code to split the createXCOFFObjectfile to `32|64`. 
If it goes through I can introduce the change for objectfilexcoff too, if we are able to introduce. 
For this one, I can change try changing to `llvm::object::ObjectFile::createObjectFile`

Reply:
Corrected it here @labath.